UefiCpuPkg/PiSmmCpuDxeSmm/PiSmmCpuDxeSmm.c

```diff
@@ -1142,8 +1149,15 @@ PiCpuSmmEntry (
   // Retrive the allocated SmmBase from gSmmBaseHobGuid. If found,
   // means the SmBase relocation has been done.
   //
-  mCpuHotPlugData.SmBase = GetSmBase (mMaxNumberOfCpus);
-  if (mCpuHotPlugData.SmBase != NULL) {
+  mCpuHotPlugData.SmBase = NULL;
+  Status                 = GetSmBase (mMaxNumberOfCpus, &mCpuHotPlugData.SmBase);
+  if (Status == EFI_OUT_OF_RESOURCES) {
+    ASSERT (Status != EFI_OUT_OF_RESOURCES);
+    CpuDeadLoop ();

```

Comment:
Doesn't need to be done in this PR, but we could consider replacing these `CpuDeadLoop()` calls with `PanicLib` in the future.

Reply:
That would make sense.  I'll leave it for a new PR just for coherency.